### DISCUSSION TITLE
IndirectLight building consumption

### DISCUSSION THREAD

**Discussion by ThomasGorisse** [google/filament#4328] [Q&A]
Hi,

We are currently investigating some memory leaks turning around buffer Images consumption with the help of ARCore team here: https://github.com/google-ar/arcore-android-sdk/issues/1262#issue-935559897

And it seems that Sceneform is kidnapping some image buffers ended with none available anymore for ARCore processing.

One of our think is that it could come from the environment cube map processed for indirect Lighting.
My first target is this call made from the frame callbacks.

My question:
### Do you think this indirect light framed called could be consuming?

```java
IndirectLight indirectLight = new IndirectLight.Builder() .reflections(reflectCubemap) .irradiance(SH_ORDER, irradianceData) .intensity(intensity * lightEstimate) .build(EngineInstance.getEngine().getFilamentEngine())
```

Sceneform indirect light is still made the "old" way and doesn't yet use the KTXLoader.

**Comment by romainguy** [comment]
When you get a new IBL from ARCore you'll have to free up the previous IBL (`IndirectLight`) by calling `Engine.destroy(IndirectLight)`, otherwise you'll leak the memory yes.

**Comment by ThomasGorisse** [comment]
Thanks for your answer.

We went on the wrong direction for solving a memory consumption.

It didn't came from the IBL but from some duplicate calls to ARCore consuming functions inside the FrameCallBack.

https://github.com/google-ar/arcore-android-sdk/issues/1262
https://github.com/ThomasGorisse/sceneform-android-sdk/pull/121

**Reply by ThomasGorisse** [reply]
By the way, We updated Filament to the 1.10.7 and UberShaderLoader and everything is looking good.
Except the still not resolved `transmission = 1.0` model issue we have on real devices:  https://github.com/ThomasGorisse/sceneform-android-sdk/discussions/76

**Reply by romainguy** [reply]
On what devices? All devices? Some devices?

**Reply by ThomasGorisse** [reply]
Pixel 4a for my part.
I will ask for other impacted devices.
But I  still didn't finished the checks to know if it doesn't come from some IBL intensity modifications which shouldn't normally happen since the test is made on our non-ar sample.
